source/connection_manager.c

```diff
@@ -638,6 +652,26 @@ static void prvUnsolicitedIncomingPublishCallback( MQTTAgentContext_t * pMqttAge
 
 /*-----------------------------------------------------------*/
 
+static void prvIncomingPublishCallback( MQTTAgentContext_t * pMqttAgentContext,
+                                        uint16_t packetId,
+                                        MQTTPublishInfo_t * pxPublishInfo )
+{
+    BaseType_t publishHandled = pdFALSE;
+
+    /* Fan out the incoming publishes to the callbacks registered using
+     * subscription manager. */
+    publishHandled = handleIncomingPublishes( pxPublishInfo );
+
+    /* If there are no callbacks to handle the incoming publishes, pass it to
+     * the unsolicited publish handler. */
+    if( publishHandled != pdTRUE )
+    {
+        prvUnsolicitedIncomingPublishCallback( pMqttAgentContext, packetId, pxPublishInfo );
```

Comment:
`prvUnsolicitedIncomingPublishCallback` is short enough that we can just paste its code here, since it isn't even a callback any more